### DISCUSSION TITLE
Muliapps with Picard iteration

### DISCUSSION THREAD

**Discussion by Xelver** [idaholab/moose#19408] [Q&A General] [answered]
Hi,

I'm now using `TransientMultiApp` to solve a "Tight Coupling" problem.  It contains a masterapp and two subapps. Normally, the order that I wished the simulation to progress is "sub1 -> master -> sub2 -> sub1 ......". So i set `execute_on  = 'initial timestep_begin'` for sub1 while `execute_on  = 'initial timestep_end'` for sub2, and I used Picard iteration to reach a stationary point at each timestep.
However, here come's the problem. I found that the values that sub1 got from sub2 and master were those of a timestep before. Is there any way to solve the problem?

Besides, I used the Picard iteration by setting `picard_max_its` in the `Executioner`  while I found that in many moose test examples, `fixed_point_max_its` was set instead of  `picard_max_its`. So I'm wondering which one is the correct way to use Picard iteration. 

Thanks a lot!

**Comment by GiudGiud** [comment, marked as answer]
Hello

you should have gotten deprecation warnings for `picard_max_its`. 
Picard is a fixed point algorithm, `fixed_point_max_its` is the new name for that. 

If you dont want sub1 to get values from the previous timestep then dont execute it at timestep_begin, execute it at timestep_end.
If you want to be converging values in sub1 with output from sub2 and iterating on this coupling, you should consider making a chain: main -> sub1 -> sub2 

Guillaume

**Reply by Xelver** [reply]
I see. Thanks! It really helps!
